File: CCC-school-work/comporg/pnc2/bschulte/ruby/primeregbs.rb
#!/usr/bin/ruby
##############

require "benchmark"

if ARGV.length < 1
	puts "Too Few Arguments"
	exit
elsif Integer(ARGV[0]) <= 1
	puts "UpperBound must be at least 2"
	exit
else
	upBound=Integer(ARGV[0])
end

divBy=2
i=3
num=2

time = Benchmark.realtime do

$stderr.printf "2 3 "

while num != upBound
	sqt=Integer(Math.sqrt(i))
	mod=i%divBy
	
	if mod != 0
		if divBy != sqt
			divBy+=1
		elsif divBy==sqt
			$stderr.printf "%d ", i
			i+=1
			num+=1
			divBy=2
		end	
	else
		i+=1
		divBy=2
	end	
end



end
$stdout.printf "%8.4f\n", time

File: CCC-school-work/comporg/pnc2/bschulte/python3/primeregbs.py
import time
import sys
import math

if(len(sys.argv) >= 2):
	if(int(sys.argv[1])<=0):
		print("Bad Argument- must be greater than 0", file=sys.stderr)
		exit(0)
else:
	print("No Upper Bound", file=sys.stderr)
	exit(0)

upBound=int(sys.argv[1])
divBy=2
i=3
num=2

start = time.time()

print("2 3 ", file=sys.stderr, end='')

while(num!=upBound):
	sqt=int(math.sqrt(i))
	mod=i%divBy

	if(mod!=0):
		if(divBy!=sqt):
			divBy+=1
		elif (divBy==sqt):
			print("%d " % i, file=sys.stderr, end='')
			i+=1
			num+=1
			divBy=2
		else:
			print("%d " % i, file=sys.stderr, end='')
			i+=1
			num+=1
			divBy=2
	else:
		i+=1
		divBy=2



end = time.time()
print("%8.4f" % (end-start))
